java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/amqp/ConnectionHandler.java

```diff
@@ -22,27 +28,26 @@
 
 	private static final Logger TRACE_LOGGER = Logger.getLogger(ClientConstants.SERVICEBUS_CLIENT_TRACE);
 
-	private final String hostname;
 	private final String username;
 	private final String password;
 	private final IAmqpConnection messagingFactory;
 
-	public ConnectionHandler(final IAmqpConnection messagingFactory, final String hostname, final String username, final String password)
+	public ConnectionHandler(final IAmqpConnection messagingFactory, final String username, final String password)
 	{
 		add(new Handshaker());
 
-		this.hostname = hostname;
 		this.username = username;
 		this.password = password;
 		this.messagingFactory = messagingFactory;
 	}
-
+	
 	@Override
 	public void onConnectionInit(Event event)
 	{
-		Connection connection = event.getConnection();
-		connection.setHostname(this.hostname + ":" + ClientConstants.AMQPS_PORT);
-		connection.setContainer(UUID.randomUUID().toString());
+		final Connection connection = event.getConnection();
+		final String hostName = event.getReactor().getConnectionAddress(connection);
+		connection.setHostname(hostName);
```

Comment:
>connection.setHostname(hostName) [](start = 2, length = 32)

Pure AMQP clients wouldn't need this step. This is EventHubs service requirement to add hostname property to Open performative - to infer namespace.